src/test/java/com/artipie/asto/FileStorageTest.java

```diff
@@ -125,4 +125,30 @@ void move(@TempDir final Path tmp) {
         storage.move(source, destination);
         MatcherAssert.assertThat(storage.value(destination), Matchers.equalTo(data));
     }
+
+    @Test
+    void shouldExistForSavedKey(@TempDir final Path tmp) {
+        final BlockingStorage storage = new BlockingStorage(new FileStorage(tmp));
+        final Key key = new Key.From("some", "key");
+        final byte[] data = "some data".getBytes();
+        storage.save(key, data);
+        MatcherAssert.assertThat(storage.exists(key), Matchers.equalTo(true));
+    }
+
+    @Test
+    void shouldNotExistForUnknownKey(@TempDir final Path tmp) {
+        final BlockingStorage storage = new BlockingStorage(new FileStorage(tmp));
+        final Key key = new Key.From("unknown");
+        MatcherAssert.assertThat(storage.exists(key), Matchers.equalTo(false));
+    }
+
+    @Test
+    void shouldNotExistForParentOfSavedKey(@TempDir final Path tmp) {
+        final BlockingStorage storage = new BlockingStorage(new FileStorage(tmp));
+        final Key parent = new Key.From("a", "b");
+        final Key key = new Key.From(parent, "c");
+        final byte[] data = "content".getBytes();
```

Comment:
@olegmoz `data` can be inlined here, `key` too, but in this test case I'd suggest to leave it as is

Reply:
@olegmoz what about this one?

Reply:
@olenagerasimova well, in general I do not think that is inlining varibales is a good idea, that is why I write code the way I do. So of course I'd prefer to do no changes, when you do suggest to keep it as it is.

Reply:
@olegmoz hm, my comment was unclear: I suggested to inline `data` and do not inline `key`. 
I personally do not like not inlined variables: they always make me think about why are they not inlined and look extra carefully for usages, which means waste of time. But since this project still has no code style rules/recommendations, let's leave it as is.